cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/runtime/spi/runtimejob/TetheringRuntimeJobManager.java

```diff
@@ -0,0 +1,137 @@
+import io.cdap.cdap.api.common.Bytes;
+import io.cdap.cdap.api.messaging.MessagePublisher;
+import io.cdap.cdap.api.messaging.TopicNotFoundException;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.internal.tethering.TetheringControlMessage;
+import io.cdap.cdap.internal.tethering.runtime.spi.provisioner.TetheringConf;
+import io.cdap.cdap.internal.tethering.runtime.spi.provisioner.TetheringProvisioner;
+import io.cdap.cdap.messaging.MessagingService;
+import io.cdap.cdap.messaging.context.MultiThreadMessagingContext;
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.cdap.proto.id.TopicId;
+import io.cdap.cdap.runtime.spi.ProgramRunInfo;
+import io.cdap.cdap.runtime.spi.runtimejob.RuntimeJobDetail;
+import io.cdap.cdap.runtime.spi.runtimejob.RuntimeJobInfo;
+import io.cdap.cdap.runtime.spi.runtimejob.RuntimeJobManager;
+import io.cdap.cdap.runtime.spi.runtimejob.RuntimeJobStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Tethering runtime job manager. This class sends runtime job details to the control channel, which will be passed
+ * along for the tethered CDAP instance to run.
+ * An instance of this class is created by {@link TetheringProvisioner}.
+ */
+public class TetheringRuntimeJobManager implements RuntimeJobManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TetheringRuntimeJobManager.class);
+  private static final Gson GSON = new GsonBuilder().create();
+
+  private final String tetheredInstanceName;
+  private final String tetheredNamespace;
+  private final CConfiguration cConf;
+  private final MessagePublisher messagePublisher;
+
+  public TetheringRuntimeJobManager(TetheringConf conf, CConfiguration cConf, MessagingService messagingService) {
+    this.tetheredInstanceName = conf.getTetheredInstanceName();
+    this.tetheredNamespace = conf.getTetheredNamespace();
+    this.cConf = cConf;
+    this.messagePublisher = new MultiThreadMessagingContext(messagingService).getMessagePublisher();
+  }
+
+  @Override
+  public void launch(RuntimeJobInfo runtimeJobInfo) throws Exception {
+    ProgramRunInfo runInfo = runtimeJobInfo.getProgramRunInfo();
+    LOG.debug("Launching run {} with following configurations: tethered instance name {}, tethered namespace {}.",
+              runInfo.getRun(), tetheredInstanceName, tetheredNamespace);
+    byte[] payload = Bytes.toBytes(GSON.toJson(createPayload()));
+    TetheringControlMessage message = new TetheringControlMessage(TetheringControlMessage.Type.RUN_PIPELINE, payload);
+    publishToControlChannel(message);
+  }
+
+  @Override
+  public Optional<RuntimeJobDetail> getDetail(ProgramRunInfo programRunInfo) {
+    // TODO: pull job status instead of always treating it as RUNNING
```

Comment:
nit: could you please file a jira for this for tracking purpose and add it as // TODO: CDAP-<jira-number> <comment>